StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1625ElementDocumentationMustNotBeCopiedAndPasted.cs

```diff
@@ -75,7 +80,40 @@ internal class SA1625ElementDocumentationMustNotBeCopiedAndPasted : DiagnosticAn
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterCompilationStartAction(HandleCompilationStart);
+        }
+
+        private static void HandleCompilationStart(CompilationStartAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleDocumentationTrivia, SyntaxKind.SingleLineDocumentationCommentTrivia);
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleDocumentationTrivia, SyntaxKind.MultiLineDocumentationCommentTrivia);
+        }
+
+        private static void HandleDocumentationTrivia(SyntaxNodeAnalysisContext context)
+        {
+            DocumentationCommentTriviaSyntax syntax = context.Node as DocumentationCommentTriviaSyntax;
+
+            HashSet<string> documentationTexts = new HashSet<string>();
```

Comment:
I haven't studied the PR trends enough to recognize that this is particularly more expensive than average. Is it that we create a HashSet<T> and we're concerned about GC pressure, or is it CPU time that you expect to suffer?
We could mitigate the HashSet<T> GC pressure by creating a thread-safe pool of HashSet<T> objects that your analyzers can 'borrow' from.